web/static/js/components/group_label_input.jsx

```diff
@@ -7,23 +7,63 @@ import styles from "./css_modules/group_label_input.css"
 // exceed the 1/4th column width on desktop, even with the widest possible characters
 const MAX_LENGTH_OF_GROUP_NAME = 20
 
-const GroupLabelInput = ({ groupWithAssociatedIdeasAndVotes, actions }) => (
-  <div className="ui fluid input">
-    <input
-      type="text"
-      className={styles.textInput}
-      placeholder="Add a group label"
-      defaultValue={groupWithAssociatedIdeasAndVotes.label}
-      maxLength={MAX_LENGTH_OF_GROUP_NAME}
-      onBlur={e => {
-        actions.submitGroupLabelChanges(groupWithAssociatedIdeasAndVotes, e.target.value)
-      }}
-    />
-    <div className="instruction">
-      <span className="keyboard-key">tab</span> to submit
-    </div>
-  </div>
-)
+class GroupLabelInput extends Component {
+  state = (() => {
+    const { groupWithAssociatedIdeasAndVotes } = this.props
+    const { label } = groupWithAssociatedIdeasAndVotes
+    return {
+      oldLabel: label,
+      showCheckmarkForLabelId: null,
+    }
+  })();
+
+  static getDerivedStateFromProps(props, state) {
+    const { oldLabel } = state
+    const { groupWithAssociatedIdeasAndVotes } = props
+    const newLabel = groupWithAssociatedIdeasAndVotes.label
+    if (oldLabel !== newLabel) {
+      return ({
+        oldLabel: newLabel,
+        showCheckmarkForLabelId: groupWithAssociatedIdeasAndVotes.id })
```

Comment:
I think we can get away with changing `showCheckmarkForLabelId` -> `showCheckmarkForLabel` (Boolean), and assigning `true` in this case, and setting `false` in the constructor, as the `group`'s id will be unique to this instance.

Reply:
We initially added the label id check since we found that all components were being updated when only one's label was modified.  However, with the new code, this no longer happens.  Thanks.  Modifying as per suggestion.